src/lib.rs

```diff
@@ -90,173 +90,185 @@ impl TryFrom<Ident> for Traits {
 }
 
 impl Traits {
-    fn type_(&self) -> Type {
+    fn type_(self) -> Type {
         use Traits::*;
 
         syn::parse_str(match self {
             Clone => "::core::clone::Clone",
-            /*Debug => "::core::fmt::Debug",
+            Debug => "::core::fmt::Debug",
             Eq => "::core::cmp::Eq",
             Hash => "::core::hash::Hash",
             PartialEq => "::core::cmp::PartialEq",
             PartialOrd => "::core::cmp::PartialOrd",
-            Ord => "::core::cmp::Ord",*/
+            Ord => "::core::cmp::Ord",
         })
         .expect("couldn't pass path to trait")
     }
 
-    fn generate_body(&self, data: &Data) -> TokenStream {
+    fn generate_body(self, name: &str, data: &Data) -> TokenStream {
```

Comment:
```suggestion
    fn generate_body(self, name: &Ident, data: &Data) -> TokenStream {
```
Is there  a reason to use a string instead?

Reply:
Yes, the name is only used by the `Debug` implementation, it needs a string.

Reply:
Has been done.